omnibus/files/private-chef-upgrades/001/037_elasticsearch_to_opensearch_migration.rb

```diff
@@ -0,0 +1,38 @@
+define_upgrade do
+    if Partybus.config.bootstrap_server
+  
+      es = Partybus.config.running_server["private_chef"]["opensearch"]
+      # Run these migrations only if elasticsearch has been enabled in the config
+      # We do not want to run this if solr is enabled
+  
+      if es["enable"]
+  
+        must_be_data_master
+
+        sleep 30
+  
+        log "All orgs are in the 503 mode..."
+        log "Migrating indexed search data..."
+        if Dir.exist?("/var/opt/opscode/elasticsearch/data/")
+          # Make sure API is down
+          stop_services(["nginx", "opscode-erchef", "opensearch"])
+          run_command("cp -r /var/opt/opscode/elasticsearch/data/ /var/opt/opscode/opensearch/data/")
+        else
+          start_services(["opensearch",
+            "nginx",
+            "oc_bifrost",
+            "opscode-erchef",
+            "postgresql",
+            "redis_lb"])
+          run_command("chef-server-ctl reindex -a -d -t")
+          stop_services(["opensearch",
+            "nginx",
+            "oc_bifrost",
+            "opscode-erchef",
+            "redis_lb"])
```

Comment:
I think starting postgresql in a no-op is the service does not exist as in the case of external postgresql.

@jashaik are there failures if we stop it in case of external pg? or any failures in reconfigure?

Reply:
Yes, Prajakta.Not including postgres in services here since external postgres cannot be managed via chef-server-ctl